backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/monitoring/VmsMonitoring.java

```diff
@@ -326,6 +331,16 @@ private void saveVmStatistics(List<VmAnalyzer> vmAnalyzers) {
         });
     }
 
+    private void clearVmNuma(List<VmAnalyzer> vmAnalyzers) {
+        List<Guid> vmIds = vmAnalyzers.stream()
+                .filter(vmAnalayzer -> vmAnalayzer.isPossibleNumaClear() &&
+                        vmAnalayzer.getVmManager().getCpuPinningPolicy() == CpuPinningPolicy.RESIZE_AND_PIN_NUMA)
```

Comment:
it would be nicer if this logic is encapsulated in VmAnalyzer so VmsMonitoring would just ask for the VMs that need to be updated (possibly with filtering out of null values or some other mechanism to filter out vms that shouldn't be updated)

Reply:
Hopefully I hit your request :)